src/vs/platform/quickinput/browser/commandsQuickAccess.ts

```diff
@@ -334,17 +344,15 @@ export class CommandsHistory extends Disposable {
 			return;
 		}
 
-		this.configuredCommandsHistoryLength = CommandsHistory.getConfiguredCommandHistoryLength(this.configurationService);
+		this.configuredCommandsHistoryLength = this.getConfiguredLength();
 
-		if (CommandsHistory.cache && CommandsHistory.cache.limit !== this.configuredCommandsHistoryLength) {
-			CommandsHistory.cache.limit = this.configuredCommandsHistoryLength;
-
-			CommandsHistory.saveState(this.storageService);
```

Comment:
with the removal of this line, I don't see when `save` (fka `saveState`) is called in the Standalone case. Not entirely sure what that did in that case, but if it did _something_ this PR causes state saving to not happen.

Reply:
I belieive in standalone case we do not have persistent storage. So saving state is a noop in this case and reload will always starts fresh. But changing to @bpasero suggestion will handle this